experiment/nipopow_verifier/tests/test_submit_over_contest.py

```diff
@@ -76,13 +76,23 @@ def test_boi_in_large():
     pt = ProofTool("../data/proofs/")
     interface = make_interface(backend)
 
-    submit_proof_name, contest_proof_name, lca = pt.create_proof_and_forkproof(
-        1000, 200, 100
-    )
+    (
+        submit_proof_name,
+        contest_proof_name,
+        lca,
+        header,
+        header_map,
+        interlink_map,
+    ) = pt.create_proof_and_forkproof(1000, 200, 100)
     submit_proof = Proof()
     contest_proof = Proof()
     submit_proof.set(pt.fetch_proof(submit_proof_name))
-    contest_proof.set(pt.fetch_proof(contest_proof_name))
+    contest_proof.set(
+        pt.fetch_proof(contest_proof_name),
+        header=header,
+        header_map=header_map,
+        interlink_map=interlink_map,
+    )
```

Comment:
Why is it necessary this is repeated? It's in almost all of the following files too.

Reply:
Refactored